docs/cilium.md

```diff
@@ -11,3 +11,53 @@ Hence, in this configuration in Kubespray, Cilium will always contact
 the external loadbalancer (even from a node in the control plane)
 and if there is no external load balancer It will ignore any local load
 balancer deployed by Kubespray and **only contacts the first master**.
+
+## Choose Cilium version
+
+```yml
+cilium_version: v1.8.8 ## or 1.9.5
+```
+
+## Add variable to config
+
+Use following variables:
+
+Example:
+
+```yml
+cilium_config_extra_vars:
+  enable-endpoint-routes: true
+```
+
+## Install Cilium Hubble
+
+k8s-net-cilium.yml:
+
+```yml
+cilium_enable_hubble: true ## enable support hubble in cilium
+cilium_hubble_install: true ## install hubble-relay, hubble-ui
```

Comment:
just a question: Can I know why we need to have 2 different options(cilium_enable_hubble and cilium_hubble_install) for enabling hubble and installing hubble?
Is there any actual use case to specify different values on those options?

Reply:
cilium_enable_hubble - enable grpc hubble service in cilium. Cilium opens a port (4244) and listens to it.
https://github.com/cilium/cilium/blob/v1.9.5/install/kubernetes/experimental-install.yaml#L161
If  cilium_hubble_install: true, then hubble components will be installed (hubble-relay, hubble-ui, etc..)
If cilium_enable_hubble = false, then cilium_hubble_install should not run because the hubble-relay and hubble-ui won't work 
If we use other hubble installation, cilium_hubble_install - not required, but cilium_enable_hubble must be true

Reply:
This playbook runs just after deploying Kubernetes cluster.
So how to use the other hubble installation when cilium_enable_hubble is true?

I feel it is straightforward to install hubble components when cilium_enable_hubble is true.

Reply:
@oomichi I even agree with you. but this feature was included earlier. Perhaps they did not fully use the hubble, but only used metrics. I propose to leave this opportunity. After all, kubespray is a universal product.

Reply:
@oomichi how are you? 😁

Reply:
@bac-w 

> this feature was included earlier. Perhaps they did not fully use the hubble, but only used metrics. I propose to leave this opportunity. After all, kubespray is a universal product.

Sorry, I could not follow what the above means?
Coujd you rephrase ?

Reply:
@oomichi 
I meant that the changes that were added earlier allow for more universal use of kubespray.
cilium_enable_hubble allows you to enable hubble for example, getting metrics.
cilium_hubble_install allows you to install traffic visualization components. hubble_ui, hubble_relay and more. I suggest leaving this option for those who do not need these components.

Reply:
Does it mean cilium_hubble_install installs additional components than hubble-metrics?
Is it correct that hubble-metrics is installed even if cilium_hubble_install is false?

Reply:
> Does it mean cilium_hubble_install installs additional components than hubble-metrics?
> Is it correct that hubble-metrics is installed even if cilium_hubble_install is false?

Hubble metrics it's not additional component, it's options for hubble component in cilium.
When enable hubble in cilium config (cilium_enable_hubble), then hubble launched with the cilium demon.

![изображение](https://user-images.githubusercontent.com/57893632/116196468-3987a780-a73c-11eb-8179-f1d3d9fbae0d.png)

And listening port 4244/tcp (by default)

![изображение](https://user-images.githubusercontent.com/57893632/116196687-7d7aac80-a73c-11eb-82fb-dc32aeb8eb7f.png)

After that, you can connect to hubble from anywhere, in your network by port 4244.
If you not use metrics, you do disable them, cilium_enable_hubble_metrics=false (default: false) or cilium_enable_hubble_metrics=true and choose the metrics you need via cilium_hubble_metrics

![изображение](https://user-images.githubusercontent.com/57893632/116199385-ba946e00-a73f-11eb-8f66-c4dbeedb1f52.png)


You can read about hubble here https://docs.cilium.io/en/v1.9/intro/#what-is-hubble and you can read about hubble metrics here https://docs.cilium.io/en/v1.9/operations/metrics/#hubble-metrics.
Options for hubble in cilium roles taken from official cilium helm chart and used for different cases.

> Does it mean cilium_hubble_install installs additional components than hubble-metrics?

Yes, in some cases these components are not needed.

> Is it correct that hubble-metrics is installed even if cilium_hubble_install is false?

Yes, but need to cilium_enable_hubble_metrics is true

@oomichi I hope now it is more clear =))

Reply:
@bac-w Thanks for your explanation, that is clear :-)

/lgtm